internal/cmd/instance/update_sharing_instance.go

```diff
@@ -0,0 +1,129 @@
+	return &UpdateSharingCmd{Context: context, share: share}
+}
+
+// Prepare returns cobra command
+func (shc *UpdateSharingCmd) Prepare(prepare cmd.PrepareFunc) *cobra.Command {
+	result := &cobra.Command{
+		PreRunE: prepare(shc, shc.Context),
+		RunE:    cmd.RunE(shc),
+	}
+	if shc.share {
+		result.Use = "share-instance [name] --id service-instance-id "
+		result.Short = "Share a service instance"
+		result.Long = `Share a service instance so that it can be consumed from various platforms in your subaccount.
+Instance can be shared only if it was created with the plan that supports instance sharing. For more information, see the documentation of the service whose instance you want to share.`
+	} else {
+		result.Use = "unshare-instance [name] --id service-instance-id "
+		result.Short = "Unshare a service instance"
+		result.Long = `Unshare a service instance to disable its consumption from any but the original platform in which it was created in your subaccount. If an instance you want to unshare has references, an error is returned`
+	}
+	result.Flags().StringVarP(&shc.instanceID, "id", "", "", "Id of the instance. Required in case when there are instances with same name")
+	cmd.AddFormatFlag(result.Flags())
+	cmd.AddModeFlag(result.Flags(), "async")
+
+	return result
+}
+
+// Validate validates command's arguments
+func (shc *UpdateSharingCmd) Validate(args []string) error {
+	if len(args) < 1 {
+		return fmt.Errorf("service instance name is required")
+	}
+	shc.instanceName = args[0]
+	return nil
+}
+
+// Run runs the command's logic
+func (shc *UpdateSharingCmd) Run() error {
+	if shc.instanceID == "" {
+		instances, err := shc.Client.ListInstances(&query.Parameters{
+			FieldQuery: []string{
+				fmt.Sprintf("name eq '%s'", shc.instanceName),
+			},
+			GeneralParams: shc.Parameters.GeneralParams,
+		})
+		if err != nil {
+			return err
+		}
+		if len(instances.ServiceInstances) == 0 {
+			return fmt.Errorf("no instances found with name %s", shc.instanceName)
+		}
+
+		if len(instances.ServiceInstances) > 1 {
+			return fmt.Errorf("more than 1 instance found with name %s. Use --id flag to specify one", shc.instanceName)
+		}
+
+		shc.instanceID = instances.ServiceInstances[0].ID
+	}
+	resultInstance, location, err := shc.Client.UpdateInstance(shc.instanceID, &types.ServiceInstance{
+		Shared: shc.share,
+	}, nil)
+	var message = ""
+	if err != nil {
+		if shc.share {
+			message = "Could not share service instance. Reason: "
+		} else {
+			message = "Could not unshare service instance. Reason: "
+		}
+		output.PrintMessage(shc.Output, message)
+		return err
+	}
+
+	if len(location) != 0 {
+		if shc.share {
+			message = fmt.Sprintf("Service Instance %s successfully scheduled for sharing. To see status of the operation use:\n", shc.instanceName)
+
+		} else {
+			message = fmt.Sprintf("Service Instance %s successfully scheduled for unsharing. To see status of the operation use:\n", shc.instanceName)
```

Comment:
Service instance "%s" successfully scheduled for unsharing. To see the status of the operation, use: \n, 

Same question as above for sharing.

Reply:
service name

Reply:
Ok, then we need " "